### PR TITLE
The scope of the unsafe blocks can be appropriately reduced

### PR DESCRIPTION
**Pull request by peamaeq**

### CODE REVIEW

**Comment by Ralith** [comment]
Thanks! Please run `cargo fmt`.

**Comment by peamaeq** [comment]
> Thanks! Please run `cargo fmt`.
OK✔

File: src/world.rs [hunk 1]
```diff
@@ -1148,17 +1146,17 @@ where
     fn next(&mut self) -> Option<Entity> {
         let components = self.inner.next()?;
         let entity = self.entities.alloc();
+        let index = unsafe { self.archetype.allocate(entity.id) };
         unsafe {
-            let index = self.archetype.allocate(entity.id);
             components.put(|ptr, ty| {
```
Comment (Ralith) [inline src/world.rs:1151]: This isn't an improvement.

**Review by Ralith — CHANGES_REQUESTED** [review]

File: src/world.rs (hunk 1 above)
Comment (Ralith) [reply src/world.rs:1151]: Oh, I see the point, nevermind.

File: src/archetype.rs [hunk 2]
```diff
@@ -300,10 +299,9 @@ impl Archetype {
                         storage,
                     }
                 })
-                .collect::<Box<[_]>>();
-
-            self.data = new_data;
-        }
+                .collect::<Box<[_]>>()
+        };
+        self.data = new_data;
```
[the remark this answers is not in the document]
Comment (peamaeq) [reply src/archetype.rs:306]: Sorry for the long reply, there is no borrowck issue here.

**Comment by Ralith** [comment]
Thanks!

**Pull request merged**